src/app/services/groupsocket/groupsocket.service.ts

```diff
@@ -65,6 +65,13 @@ export class GroupSocketService {
     this.socket.on('getExistingSelectedLocation', res => {
       this.locationSubject.next(res);
     });
+
+    this.socket.on('getLeavingUserInfo', res => {
+      const leavingUser = this.findWithAttr(this.userInfos, 'socketID', res.socketID);
+      if (leavingUser > -1) {
+        this.userInfos.splice(leavingUser, 1);
```

Comment:
I think you should be able to remove an element from an array using the `pull` function from Lodash. Check it out https://lodash.com/docs/4.17.4#pull

Reply:
I think you could accomplish what the findWithAttr method does with the javascript filter() function in one line. For example:

const leavingUser = this.userInfos.filter( x => x.socketID == res.socketID); 

This would result in leavingUser to be an array with the user whose socketID matches the one you are looking for.